### ISSUE TITLE
Feedback: API Support Policy

### ISSUE THREAD

**Issue by knz** [cockroachdb/docs#14834]
Raphael 'kena' Poss (knz) commented:

cc @taroface 
Page: https://cockroachlabs.com/docs/v22.1/api-support-policy.html

There's 3 things we need to enhance here.

### a detail I missed on the first review

> Backward-compatible changes
> The following list is not exhaustive:
> ...
> - Change in non-interactive cockroach sql shell input or output.

I'm not sure that's right?  In the *non-interactive* case, the API is stable, and so if we change the input or output in a way that breaks automation that would be a backward-incompatible change.
(But see point below for a clarification)

### need to add *definitions* for the terms "backward-incompatible change" and "backward-compatible change"

Currently the terms have *examples* but do not have *definitions*. We need to add them.

Suggestions:

- Backward-compatible change: when existing automation that relies *solely* on stable APIs continues to work as previously. If automation was relying on non-stable APIs (unstable or reserved), no guarantee is provided.

- Backward-incompatible change: when existing automation that relies *solely* on stable APIs needs to be updated to continue to work.

### Regarding the SQL shell

It's currently missing from the API table and should be there:

- the CLI flags of the SQL shell are a stable API, unless marked as unstable/preview
- the input client-side commands to the SQL shell are a stable API, unless marked as unstable/preview
- the output table formats of the SQL shell are a stable API, unless marked as unstable/preview
- any output of `cockroach sql` prefixed by `#` is an unstable API
- the behavior of `cockroach sql` when used *non-interactively*, given a set of CLI flags and output table format selection, insofar that only SQL input using stable features is executed and the `#`-prefixed output is ignored, is a stable API.
- the behavior of `cockroach sql` when used interactively is an unstable API.

In other words, it's safe to use `cockroach sql` in automation as long as the scripts only use a combination of stable features.

Jira Issue: DOC-5365
